main.go

```diff
@@ -123,24 +156,30 @@ func main() {
 		panic(err)
 	}
 
-	cnt := 0
-	for {
-		// if it's a turn for the node to publish, publish
-		if cnt == nodeId {
-			msg := fmt.Sprintf("Hello from node%d", nodeId)
-			if err := topic.Publish(ctx, []byte(msg)); err != nil {
-				log.Printf("Failed to publish: %s\n", msg)
-			} else {
-				log.Printf("Published: %s\n", msg)
-			}
+	msg := make([]byte, *msgSizeKB*(1<<10))
+	rand.Read(msg)
+
+	publishingId := rand.Intn(*countFlag)
+	dups := -1
+	// if it's a turn for the node to publish, publish
+	if publishingId == nodeId {
```

Comment:
I think the publishing node will terminate after this? because there is no forever loop.

Reply:
Ah I think I missed re-adding the stop channel while merging the code

Reply:
That's an overkill. Just make the publisher receive the messages as well.